architecture-decision-records/0024-continuous-integration-service.md

```diff
@@ -0,0 +1,44 @@
+# 24. Continuous integration service
+
+**Date:** 2022-03-15
+
+## Context
+We need a continuous integration service. There are two main functions we need from the service.
+* Test code changes before they are merged and deployed.
+* Build our code and deploy it to the various different services and environments it runs in.
+
+## Options considered
+We have been using Jenkins for a couple of years now so one option was to carry on using that.
+
+The other option we considered is using [GitHub actions](https://docs.github.com/en/actions).
+
+## Decision
+We have decided to move from Jenkins to GitHub actions. There are advantages and disadvantages to this move.
+
+### Advantages
+
+#### Server maintenance
+Jenkins runs on an EC2 instance using multiple docker images and uses multiple plugins and all of these need to be kept up to date separately which takes up quite a lot of time. 
+GitHub actions runs on a hosted service that is maintained by GitHub.
+
+#### Cost
+With Jenkins, we are paying for the EC2 instance, the load balancer and NAT gateway and the Fargate costs for running the nodes which is not insignificant. 
+GitHub actions is free for public repositories.
```

Comment:
Is it worth mentioning that the majority of the TDR repos are public, and that TDRs default position is that all repos should be public? 

Have we considered what to do for any private repos, eg tdr-configurations? Also if we need to add further private repos in the future?

Reply:
Yeah, I've mentioned our repos are mostly public. I've said that for private repos we'll have to use the organisation credits we've got.